packages/chain-adapters/src/cosmossdk/CosmosSdkBaseAdapter.ts

```diff
@@ -0,0 +1,221 @@
+/* eslint-disable @typescript-eslint/no-unused-vars */
+// no-unused-vars is temporarily disabled until more functions are implemented
+import { CAIP2, caip2, caip19 } from '@shapeshiftoss/caip'
+import { AssetNamespace } from '@shapeshiftoss/caip/dist/caip19/caip19'
+import { CosmosSignTx } from '@shapeshiftoss/hdwallet-core'
+import { BIP44Params, chainAdapters, ChainTypes } from '@shapeshiftoss/types'
+import { cosmos } from '@shapeshiftoss/unchained-client'
+import WAValidator from 'multicoin-address-validator'
+
+import { ChainAdapter as IChainAdapter } from '../api'
+import { ChainAdapter } from '../bitcoin'
+import { ErrorHandler } from '../error/ErrorHandler'
+
+export type CosmosChainTypes = ChainTypes.Cosmos | ChainTypes.Osmosis
+
+export interface ChainAdapterArgs {
+  chainId?: CAIP2
+  providers: {
+    http: cosmos.api.V1Api
+    // unchained-client 5.1.1 does not have a websocket client for cosmos
```

Comment:
Do we want to use `ws: {}` to keep it consistent?

Reply:
that doesn't provide any value. The correct way to handle this is wait for `unchained-client` to provide a websocket client for Cosmos.